src/btree/bt_compact.c

```diff
@@ -334,6 +334,11 @@ __wt_compact(WT_SESSION_IMPL *session)
         WT_STAT_CONN_INCR(session, session_table_compact_skipped);
         WT_STAT_DATA_INCR(session, btree_compact_skipped);
 
+        /* If we're an internal session we must've been called by the background compact server. */
+        if (F_ISSET(session, WT_SESSION_INTERNAL)) {
```

Comment:
We don't need the curly brackets and I remember @keitharnoldsmith suggested in another PR that it would be great to check for the session's name too, see [here](https://github.com/wiredtiger/wiredtiger/pull/9456#discussion_r1279806982).

Reply:
Do we want to increment `session_table_compact_skipped` too? I see it is still incremented line 334.

Reply:
FWIW, in that comment I was thinking we could compare the session pointer rather than the session name. I.e., `conn->background_compact.session` is the session that should be doing background compaction work.